pkg/lib/config/saml.go

```diff
@@ -104,29 +135,36 @@ func (p SAMLNameIDAttributePointer) MustGetJSONPointer() jsonpointer.T {
 }
 
 type SAMLServiceProviderConfig struct {
-	Deprecated_ID          string                     `json:"id,omitempty"`
-	ClientID               string                     `json:"client_id,omitempty"`
-	NameIDFormat           SAMLNameIDFormat           `json:"nameid_format,omitempty"`
-	NameIDAttributePointer SAMLNameIDAttributePointer `json:"nameid_attribute_pointer,omitempty"`
-	AcsURLs                []string                   `json:"acs_urls,omitempty"`
-	Destination            string                     `json:"destination,omitempty"`
-	Recipient              string                     `json:"recipient,omitempty"`
-	Audience               string                     `json:"audience,omitempty"`
-	AssertionValidDuration DurationString             `json:"assertion_valid_duration,omitempty"`
+	Deprecated_ID          string                        `json:"id,omitempty"`
+	ClientID               string                        `json:"client_id,omitempty"`
+	NameIDFormat           samlprotocol.SAMLNameIDFormat `json:"nameid_format,omitempty"`
+	NameIDAttributePointer SAMLNameIDAttributePointer    `json:"nameid_attribute_pointer,omitempty"`
+	AcsURLs                []string                      `json:"acs_urls,omitempty"`
+	Destination            string                        `json:"destination,omitempty"`
+	Recipient              string                        `json:"recipient,omitempty"`
+	Audience               string                        `json:"audience,omitempty"`
+	AssertionValidDuration DurationString                `json:"assertion_valid_duration,omitempty"`
+	SLOEnabled             bool                          `json:"slo_enabled,omitempty"`
+	SLOCallbackURL         string                        `json:"slo_callback_url,omitempty"`
+	SLOBinding             samlprotocol.SAMLBinding      `json:"slo_binding,omitempty"`
 }
 
 func (c *SAMLServiceProviderConfig) SetDefaults() {
 	if c.NameIDFormat == "" {
-		c.NameIDFormat = SAMLNameIDFormatUnspecified
+		c.NameIDFormat = samlprotocol.SAMLNameIDFormatUnspecified
 	}
 
-	if c.NameIDFormat == SAMLNameIDFormatUnspecified && c.NameIDAttributePointer == "" {
+	if c.NameIDFormat == samlprotocol.SAMLNameIDFormatUnspecified && c.NameIDAttributePointer == "" {
 		c.NameIDAttributePointer = "/sub"
 	}
 
 	if c.AssertionValidDuration == "" {
 		c.AssertionValidDuration = DurationString("20m")
 	}
+
+	if c.SLOBinding == "" {
+		c.SLOBinding = crewjamsaml.HTTPPostBinding
```

Comment:
Use samlprotocol.SAMLBindingHTTPPost. Remove the import of `crewjamsaml` in this file.